File: DemoServer/AdditionalLanguages/python/autoIndexes/autoMapIndex1/autoMapIndex1.py
from demo_example import Example, RunParamsBase
from models import Employee

class RunParams(RunParamsBase):
    def __init__(self, first_name: str):
        self.first_name = first_name

class AutoMapIndex1(Example):
    def run(self, run_params: RunParams):
        firstName = run_params.first_name

        #region Demo
        with self.document_store_holder.store().open_session() as session:
            #region Step_1
            find_employee_query = session.query(object_type=Employee).where_equals("FirstName", firstName)
            #endregion

            #region Step_2
            employee_result = find_employee_query.first()
            #endregion
        #endregion

        return employee_result

File: DemoServer/AdditionalLanguages/php/autoIndexes/autoMapIndex1/AutoMapIndex1.php
<?php

namespace RavenDB\Demo\autoIndexes\autoMapIndex1;

//region Usings
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Employee;

class AutoMapIndex1
{
    public function __invoke(RunParams $runParams): ?Employee
    {
        $firstName = $runParams->getFirstName();

        //region Demo
        $employeeResult = null;

        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
            //region Step_1
            $findEmployeeQuery = $session->query(Employee::class)
                ->whereEquals("FirstName", $firstName);
            //endregion

            //region Step_2
            $employeeResult = $findEmployeeQuery->firstOrDefault();
            //endregion
            
        } finally {
            $session->close();
        }
        //endregion

        return $employeeResult;
    }
}
